File: apps/tauri/src-tauri/src/commands/conversations.rs
use serde_json::Value;
use tauri::AppHandle;

use crate::ipc_contracts::conversations::ConversationSaveMessageRequest;

#[tauri::command]
pub fn conversations_get_or_create(app: AppHandle) -> Value {
    crate::conversations::get_or_create(&app)
}

#[tauri::command]
pub fn conversations_load_messages(app: AppHandle, conversation_id: String) -> Value {
    crate::conversations::load_messages(&app, &conversation_id)
}

#[tauri::command]
pub fn conversations_save_message(app: AppHandle, req: ConversationSaveMessageRequest) -> Value {
    let req = serde_json::to_value(&req).unwrap_or_default();
    crate::conversations::save_message(&app, &req)
}

File: apps/tauri/src/tauri-client/namespaces/conversations/conversations.ts
import { invoke } from '@tauri-apps/api/core';

import type { ConversationSaveMessageRequest } from '@ajh/shared/schemas';

export const conversations = {
  getOrCreateConversation: () => invoke('conversations_get_or_create'),
  loadMessages: ({ conversationId }: { conversationId: string }) =>
    invoke('conversations_load_messages', { conversationId }),
  saveMessage: (req: ConversationSaveMessageRequest) =>
    invoke('conversations_save_message', { req }),
  saveAllMessages: (opts: Record<string, unknown>) =>
    invoke('conversations_save_all_messages', opts),
};
